source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -21,12 +21,16 @@ namespace TransportSockets {
 namespace Tls {
 
 namespace {
+
+constexpr absl::string_view NotReadyReason{"SSL error: secret is not ready"};
```

Comment:
nit: What does this mean? Can you make the error message a bit better or potentially actually document all the potential errors for the built-in transport sockets somewhere?

Reply:
I hope now it is better, when SDS is configured there are chances that Cluster are up but Secret is not ready.